packages/address/src/address.ts

```diff
@@ -240,8 +221,7 @@ export class Address {
       );
     }
 
-    const paddedAddress = padFirst12BytesOfEvmAddress(evmAddress);
-    return new Address(paddedAddress);
+    return new Address(evmAddress);
```

Comment:
I don't understand why we'd want to have the `Address` class represent anything but a fuel-compatible address. It seems more appropriate to me that we differentiate between an `Address` and an `EvmAddress`, the same way Sway does it.

@LuizAsFight @luizstacio What are your thoughts here?

Reply:
@nedsalk I believe this is done for simplicity since on Fuel an EVM address is represented the same way as Fuel addresses, using 32 bytes. The difference is that an EVM address is padded with zeros to make it 32 bytes.